xc/common/utils/prjxray_routing_import.py

```diff
@@ -1237,6 +1253,46 @@ def find_constant_network(graph):
     return synth_tiles
 
 
+def get_pin_sides_dict(arch_file):
+    """
+    This function is used to generate a dictionary containing the sides of each pin
+    of a physical tile.
+
+    This is necessary to have a correct location of nodes when a pin is present in multiple
+    sides of a tile.
+    """
+
+    arch_xml = ET.ElementTree()
+    root_element = arch_xml.parse(arch_file)
+
+    pin_sides_dict = dict()
+    for tile in root_element.iter("tile"):
+        tile_name = tile.attrib["name"]
+        pin_sides_dict[tile_name] = dict()
+
+        for loc in tile.iter("loc"):
+            side = loc.attrib["side"]
+            if side == "left":
+                side = Sides.LEFT
+            elif side == "right":
+                side = Sides.RIGHT
+            elif side == "top":
+                side = Sides.TOP
+            elif side == "bottom":
+                side = Sides.BOTTOM
+
+            pins = loc.text.split()
+            for pin in pins:
+                pin_name = pin.split(".")[-1]
```

Comment:
Please add a comment showing an example of pin

Reply:
Done